languages/php/src/main/kotlin/io/vrap/codegen/languages/php/base/PhpBaseFileProducer.kt

```diff
@@ -2752,7 +2752,18 @@ class PhpBaseFileProducer constructor(val api: Api, @BasePackageName val package
                     |    
                     |    /** @psalm-var string */
                     |    private $!cacheKey;
-                    |
+                    |    /**
+                    |     * The constructor of the ClientCredentials class.
+                    |     *
+                    |     * @param string $!clientId
+                    |     *   The client id.
+                    |     * @param string $!clientSecret
+                    |     *   The client secret.
+                    |     * @param string $!scope
```

Comment:
What does `!` mean in this construct? `$!scope`

Reply:
it's to recognize the php variable from Kotlin

Reply:
after the compiling the variables will be without the exclamation point

Reply:
Yes. In kotlin the `$` inside of a stringtemplate is special. Only other way to escape it would be to use `${"$"}` which is a little bit weird so we made a small helper method which replaces it for the PHP use case